Lib/ftplib.py

```diff
@@ -324,11 +324,16 @@ def makeport(self):
     def makepasv(self):
         """Internal: Does the PASV or EPSV handshake -> (address, port)"""
         if self.af == socket.AF_INET:
-            untrusted_host, port = parse227(self.sendcmd('PASV'))
-            if self.trust_server_pasv_ipv4_address:
-                host = untrusted_host
-            else:
-                host = self.sock.getpeername()[0]
+            try:
+                untrusted_host, port = parse227(self.sendcmd('PASV'))
+                if self.trust_server_pasv_ipv4_address:
+                    host = untrusted_host
+                else:
+                    host = self.sock.getpeername()[0]
+            except error_reply as resp:
+                resp = str(resp)
+                if resp[:3] == '229':
```

Comment:
This is sort of breaking convention for the file, but it's also more correct. Let me know what you think.
```suggestion
                if resp.startswith('229'):
```

Reply:
We removed '229' check at all as per later suggestions. This is not applicable. Thanks.